backend/src/main/java/ba/unsa/etf/si/bbqms/ws/controllers/GroupController.java

```diff
@@ -0,0 +1,157 @@
+        }
+    }
+
+    @PostMapping("/{tenantCode}")
+    @PreAuthorize("hasAnyRole('ROLE_SUPER_ADMIN', 'ROLE_BRANCH_ADMIN')")
+    public ResponseEntity addBranchGroup(@PathVariable final String tenantCode, @RequestBody final BranchGroupCreateDto request) {
+        final User currentUser = authService.getAuthenticatedUser();
+
+        if(!currentUser.getTenant().getCode().equals(tenantCode)) {
+            return ResponseEntity.badRequest().build();
+        }
+
+        try {
+            return ResponseEntity.ok().body(BranchGroupResponseDto.fromEntity(groupService.addBranchGroup(request)));
+        } catch (Exception e) {
+            return ResponseEntity.badRequest().body(new ErrorResponseDto(e.getMessage()));
+        }
+    }
+
+    @PutMapping("/{tenantCode}/{groupId}/services/{serviceId}")
+    @PreAuthorize("hasAnyRole('ROLE_SUPER_ADMIN', 'ROLE_BRANCH_ADMIN')")
+    public ResponseEntity addBranchGroupService(@PathVariable final String tenantCode, @PathVariable final String groupId, @PathVariable final String serviceId) {
+        final User currentUser = authService.getAuthenticatedUser();
+
+        if(!currentUser.getTenant().getCode().equals(tenantCode)) {
+            return ResponseEntity.badRequest().build();
+        }
+
+        try {
+            return ResponseEntity.ok().body(BranchGroupResponseDto.fromEntity(groupService.addBranchGroupService(Long.parseLong(groupId), Long.parseLong(serviceId))));
+        } catch (Exception e) {
+            return ResponseEntity.badRequest().body(new ErrorResponseDto(e.getMessage()));
+        }
+    }
+
+    @PutMapping("/{tenantCode}/{groupId}/branches/{branchId}")
+    @PreAuthorize("hasAnyRole('ROLE_SUPER_ADMIN', 'ROLE_BRANCH_ADMIN')")
+    public ResponseEntity addBranchGroupBranch(@PathVariable final String tenantCode, @PathVariable final String groupId, @PathVariable final String branchId) {
+        final User currentUser = authService.getAuthenticatedUser();
+
+        if(!currentUser.getTenant().getCode().equals(tenantCode)) {
+            return ResponseEntity.badRequest().build();
+        }
+
+        try {
+            return ResponseEntity.ok().body(BranchGroupResponseDto.fromEntity(groupService.addBranchGroupBranch(Long.parseLong(groupId), Long.parseLong(branchId))));
+        } catch (Exception e) {
+            return ResponseEntity.badRequest().body(new ErrorResponseDto(e.getMessage()));
+        }
+    }
+
+    @PutMapping("/{tenantCode}/{branchGroupId}")
+    @PreAuthorize("hasAnyRole('ROLE_SUPER_ADMIN', 'ROLE_BRANCH_ADMIN')")
+    public ResponseEntity updateBranchGroup(@PathVariable final String tenantCode, @PathVariable final String branchGroupId, @RequestBody final BranchGroupUpdateDto name) {
+        final User currentUser = authService.getAuthenticatedUser();
+
+        if(!currentUser.getTenant().getCode().equals(tenantCode)) {
+            return ResponseEntity.badRequest().build();
+        }
+
```

Comment:
This is now written all over the codebase. Can we extract it into a method in `AuthService` like:

`boolean canChangeTenant(final String tenantCode)`,
So we can use it like:
```
if (!authService.canChangeTenant(tenantCode)) {
         return ResponseEntity.badRequest().build();
}
```